src/main/java/bio/terra/common/BaseStep.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.common;
+
+import bio.terra.model.ErrorModel;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import org.springframework.http.HttpStatus;
+
+public abstract class BaseStep implements Step {
+  @StepOutput protected Object response;
+  @StepOutput protected HttpStatus statusCode;
+
+  private FlightContext context;
+
+  @Override
+  public final StepResult doStep(FlightContext context)
+      throws InterruptedException, RetryException {
+    this.context = context;
+    StepUtils.readInputs(this, context);
+    try {
+      return perform();
+    } finally {
+      StepUtils.writeOutputs(this, context);
+    }
+  }
+
+  @Override
+  public final StepResult undoStep(FlightContext context) throws InterruptedException {
+    this.context = context;
+    StepUtils.readInputs(this, context);
+    return undo();
+  }
+
+  public abstract StepResult perform() throws InterruptedException, RetryException;
+
+  public StepResult undo() throws InterruptedException {
+    // Many steps aren't undoable.
+    return StepResult.getStepResultSuccess();
+  }
+
+  protected void setErrorResponse(String message, HttpStatus responseStatus) {
+    ErrorModel errorModel = new ErrorModel().message(message);
+    setResponse(errorModel, responseStatus);
+  }
+
+  protected void setResponse(Object responseObject, HttpStatus responseStatus) {
+    response = responseObject;
+    statusCode = responseStatus;
+  }
+
+  protected FlightContext getContext() {
```

Comment:
Since the most common use of `FlightContext` was to get at the step inputs, I wanted to try removing it as a step argument and having a method call to get at it. Once more steps are converted I think more common patterns will emerge that could be moved into a common base class.